### PR TITLE
Feature: Add HTTP2 support.

### PR DESCRIPTION
**Pull request by eloycoto**
This commit adds HTTP2 termination support to APIcast.

Things that were changed:

- ngx.location.capture: no longer used, it has no support to HTTP2,
added support will be easy in C code, but HTTP client using cosockets is
the prefered way from code maintainers and it has the same performance.

- Integration test: Changed some of them that were not correctly defined.

This commit only add support to terminate HTTP2 request, full HTTP2
support is not yet enabled.

### CODE REVIEW

**Comment by eloycoto** [comment]
@jsenko 

Around test that I added in this feature:

HTTP 2.0 termination:
    - Simple SSL 1.1 termination tests are in there. 
    - A simple test to validate that if the user request HTTP2 it replies correctly. 

HTTP 2.0 full flow:
    - HTTP2 response from endpoint. 
    - HTTP2 response from endpoint with modification in the headers. 
    - HTTP2 response from endpoint using MTLS. 

All other policies should work as normal, no protocol or code issues that I can
think about producing a new issue. Any other test to add?

File: gateway/src/apicast/policy/grpc/apicast-policy.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "http://apicast.io/policy-v1.1/schema#manifest#",
+  "name": "HTTP2 endpoint",
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply gateway/src/apicast/policy/grpc/apicast-policy.json]: It's the same way as Nginx follows, I will prefer to use grpc

File: t/fixtures/server.crt [hunk 2]
```diff
@@ -1,10 +1,13 @@
 -----BEGIN CERTIFICATE-----
-MIIBXjCCAQQCCQCILyC7QjBFsDAKBggqhkjOPQQDAjA3MQ8wDQYDVQQKDAYzc2Nh
-bGUxEDAOBgNVBAsMB0FQSWNhc3QxEjAQBgNVBAMMCWxvY2FsaG9zdDAeFw0xODA5
-MDYxMjA3MDlaFw0yODA5MDMxMjA3MDlaMDcxDzANBgNVBAoMBjNzY2FsZTEQMA4G
-A1UECwwHQVBJY2FzdDESMBAGA1UEAwwJbG9jYWxob3N0MFkwEwYHKoZIzj0CAQYI
-KoZIzj0DAQcDQgAEhkmo6Xp/9W9cGaoGFU7TaBFXOUkZxYbGXQfxyZZucIQPt89+
-4r1cbx0wVEzbYK5wRb7UiWhvvvYDltIzsD75vjAKBggqhkjOPQQDAgNIADBFAiB5
-8nGG6rkQVUtJu1Hw6SZvQBkq7bkA+rKxw7zg2WlfzwIhAKr9n17c9yiJiSd8eItm
-tVuWrmRObJCfxdMjViLP7NCF
+MIIB3TCCAYKgAwIBAgIUFc7TeXpi35a+Jqv8sCDGm6ZqhogwCgYIKoZIzj0EAwIw
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply t/fixtures/server.crt]: these certs are not managed by that Makefile, so I do not think that it's worth to add in that. Commit is self-explanatory.

[the remark this answers is not in the document]
Comment (davidor) [reply t/fixtures/server.crt]: :+1:

File: gateway/src/apicast/policy/grpc/apicast-policy.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidor) [reply gateway/src/apicast/policy/grpc/apicast-policy.json]: :+1:

File: gateway/src/apicast/backend_client.lua [hunk 3]
```diff
@@ -47,7 +38,7 @@ local function detect_http_client(endpoint)
   if proxy then -- use default client
     return
   else
-    return http_ng_backend_phase[ngx.get_phase()]
+    return http_ng_resty
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply gateway/src/apicast/backend_client.lua]: It's using lua coroutines and it's the recommended way in Openresty, it's safe to update.

**Pull request merged**